File: Python3/Minimum Number of Moves to Seat Everyone.py
#1
class Solution:
    def minMovesToSeat(self, seats: List[int], students: List[int]) -> int:
        seats.sort()
        students.sort()
        return sum(abs(e - t) for e, t in zip(seats, students))

#2
class Solution:
    def minMovesToSeat(self, seats: List[int], students: List[int]) -> int:
        seats_cnt, students_cnt = [0] * (max(seats) + 1), [0] * (max(students) + 1)
        for seat in seats:
            seats_cnt[seat] += 1
        for student in students:
            students_cnt[student] += 1
        ans = 0
        i = j = 1
        while i < len(students_cnt):
            if students_cnt[i]:
                # find the next available seat
                while j < len(seats_cnt) and not seats_cnt[j]:
                    j += 1
                ans += abs(i - j)
                seats_cnt[j] -= 1
                students_cnt[i] -= 1
            else:
                i += 1
        return ans

File: Golang/Minimum Number of Moves to Seat Everyone.go
func minMovesToSeat(seats []int, students []int) int {
    ans := 0
    sort.Ints(seats)
    sort.Ints(students)
    for i := 0; i < len(seats); i++ {
        ans += abs(seats[i] - students[i])
    }
    return ans
}

func abs(a int) int {
    if a < 0 {
        a = -a
    }
    return a
}